### ISSUE TITLE
Save mapping of which device tenants are stored on in v1/v2

### ISSUE THREAD

**Issue by mdurrant-b3** [a10networks/a10-neutron-lbaas#113]
We need to know which device a tenant "lives on" for scaling purposes.

 - [ ] Make migration for database table to store which device each tenant is on
 - [ ] Initially populate table with the tenant's current device

**Comment by Cedev** [comment]
We need to store which device each loadbalancer/vip is on. Break the indirect dependency

```                                
loadbalancer ---> tenant ---> device
              |           ^ hash of tenant id
              ^ neutron table
```

And replace it with

```
loadbalancer ---> device
              ^ new table
```

(The loadbalancer will still know which tenant it belongs to)

**Comment by Cedev** [comment]
The plan is to make one [polymorphic](http://docs.sqlalchemy.org/en/rel_1_0/orm/inheritance.html) table for loadbalancers (`A10_SLB`) and another one for devices (`A10_DEVICES`). In the alembic migrations there'll be [separate branches](http://alembic.readthedocs.org/en/latest/branches.html) for tables that depend on lbaas v1 and lbaas v2. The SQL alchemy classes for the version specific models will only be defined `if` there's support for that version.

![a10_lb](https://cloud.githubusercontent.com/assets/1644042/10526490/72f72aa0-7348-11e5-8d31-823d9cbf37df.png)

**Comment by Cedev** [comment]
To do:

 - [ ] Rename `device_key` to `configuration_key`
 - [ ] Make migration to populate existing appliances (devices)
 - [ ] Make migration to populate appliance (device) for existing lbaas v1 vips
 - [ ] Make migration to populate appliance (device) for existing lbaas v2 loadbalancers
 - [ ] Create appliances as needed
 - [ ] Create `a10_slb_v1` when handling a v1 `vip` creation
 - [ ] Delete the `a10_slb_v1` when handling a v1 `vip` removal
 - [ ] Create `a10_slb_v2` when handling a v2 `loadbalancer` creation
 - [ ] Delete the `a10_slb_v2` when handling a v2 `loadbalancer` removal
 - [ ] patch whatever `vip` code is checking the hash to get the device to instead get it from the model (will still use the hash to pick devices for new vips)
 - [ ] patch whatever `loadbalancer` code is checking the hash to get the device to instead get if from the model (will still use the hash to pick devices for new loadbalancers)
 - [ ] Make the `A10ApplianceConfigured` model present the data from the configuration (might be a separate story)

**Comment by Cedev** [comment]
To do the migration we need to load the real configuration. I tried hacking one together by making an `A10OpenstackLBV1(None)`, but that doesn't get the provider name so we don't know which `vips` are ours. We're going to have to figure out how to load the real deal, enumerate the providers, and check to see if they are the a10 provider we're working with.

**Comment by Cedev** [comment]
The following demonstrates getting the (almost) real configuration for the drivers. It doesn't provide a `plugin` and doesn't import all of the set up for oslo configuration that is set up when a driver loads for real.

```
import neutron.plugins.common.constants as constants
import neutron.services.service_base as service_base
from oslo_config import cfg

cfg.CONF(project='neutron')

lbv2 = service_base.load_drivers(constants.LOADBALANCERV2,None)
print lbv2
print lbv2[0][lbv2[1]].a10.hooks.select_device('012-345')
```

The migrations are going to use this mechanism to get the drivers and their provider names.

**Comment by Cedev** [comment]
The cli unfortunately includes a workaround for [https://bitbucket.org/zzzeek/alembic/issues/336/branching-causes-destination-is-not-a](alambic issue 336).

**Comment by Cedev** [comment]
Blocked by #122
